dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassPathBuilder.java

```diff
@@ -70,39 +65,19 @@ public ClassPathResult resolve(List<Artifact> artifacts) {
     DependencyGraphResult result = dependencyGraphBuilder.buildFullDependencyGraph(artifacts);
     List<DependencyPath> dependencyPaths = result.getDependencyGraph().list();
 
+    // TODO should DependencyGraphResult have a mediate() method that returns a ClassPathResult?
+    
     // To remove duplicates on (groupId:artifactId) for dependency mediation
-    Map<String, String> keyToFirstArtifactVersion = Maps.newHashMap();
+    DependencyMediation mediation = new DependencyMediation();
 
     for (DependencyPath dependencyPath : dependencyPaths) {
       Artifact artifact = dependencyPath.getLeaf();
-      File file = artifact.getFile();
-      if (file == null) {
-        // When artifact was not downloaded, it's recorded in result.getArtifactProblems().
-        continue;
+      mediation.put(dependencyPath);
+      if (mediation.selects(artifact)) {
```

Comment:
It's a bit odd that DependencyMediation class does not have logic to traverse a tree. Because of that, its `put` and `selects` interface cannot implement Gradle's highest-win dependency mediation.

(I don't expect this PR to address that concern; just writing down what I felt)

Reply:
renamed "MavenDependencyMediation"